circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala

```diff
@@ -0,0 +1,147 @@
+package org.http4s.circe.middleware
+
+import cats._
+import cats.effect._
+import cats.data._
+import io.circe._
+import io.circe.syntax._
+import org.http4s._
+import org.http4s.headers.Connection
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.implicits._
+import org.http4s.circe._
+
+object JsonDebugErrorHandler {
+
+  private[this] val messageFailureLogger =
+    org.log4s.getLogger("org.http4s.circe.middleware.jsondebugerrorhandler.message-failures")
+  private[this] val serviceErrorLogger =
+    org.log4s.getLogger("org.http4s.circe.middleware.jsondebugerrorhandler.service-errors")
+
+  // Can be parametric on my other PR is merged.
+  def apply[F[_]: Sync, G[_]: Applicative](
+      service: Kleisli[F, Request[G], Response[G]],
+      redactWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains
+  ): Kleisli[F, Request[G], Response[G]] = Kleisli { req =>
+    import cats.implicits._
+    implicit def entEnc[M[_]: Applicative, N[_]] = JsonErrorHandlerResponse.entEnc[M, N](redactWhen)
+
+    service
+      .run(req)
+      .handleErrorWith {
+        case mf: MessageFailure =>
+          messageFailureLogger.debug(mf)(
+            s"""Message failure handling request: ${req.method} ${req.pathInfo} from ${req.remoteAddr
+              .getOrElse("<unknown>")}""")
+          mf.inHttpResponse[F, G](req.httpVersion).map { firstResp =>
+            Response[G](
+              status = firstResp.status,
+              httpVersion = firstResp.httpVersion,
+              headers = firstResp.headers.redactSensitive(redactWhen)
+            ).withEntity(JsonErrorHandlerResponse[G](req, mf))
+          }
+        case scala.util.control.NonFatal(t) =>
```

Comment:
I don't think we need this anymore now we are off task. This had escaped my vision until this exact instance, but I bet this is in other spots as we..

```suggestion
        case t =>
```